cpp/oneapi/dal/backend/primitives/ndarray.hpp

```diff
@@ -277,6 +351,82 @@ class ndview : public ndarray_base<axis_count, order> {
     bool data_is_mutable_;
 };
 
+#ifdef ONEDAL_DATA_PARALLEL
+template <typename T1, ndorder ord1, typename T2, ndorder ord2>
+sycl::event copy_by_value(sycl::queue& q,
+                          ndview<T1, 2, ord1>& dst,
+                          const ndview<T2, 2, ord2>& src,
+                          const event_vector& deps = {}) {
+    ndshape<2> dst_shape = dst.get_shape();
+    ONEDAL_ASSERT(dst_shape == src.get_shape());
+    ONEDAL_ASSERT(dst.has_mutable_data() && src.has_data());
+    const auto cp_range = make_range_2d(dst_shape[0], dst_shape[1]);
+    if constexpr (ord1 == ndorder::c) {
+        T1* const dst_ptr = dst.get_mutable_data();
+        const T2* const src_ptr = src.get_data();
+        const auto dst_stride = dst.get_leading_stride();
+        const auto src_stride = src.get_leading_stride();
+        return q.submit([&](sycl::handler& h) {
```

Comment:
declare sycl::event before if and assign it here and in else branch, -> return it at the end of func without any ASSERT(false);

Reply:
fixed